base/applications/rapps/unattended.cpp

```diff
@@ -253,15 +253,20 @@ ParseCmdAndExecute(LPWSTR lpCmdLine, BOOL bIsFirstLaunch, int nCmdShow)
         hMutex = CreateMutexW(NULL, FALSE, szWindowClass);
         if ((!hMutex) || (GetLastError() == ERROR_ALREADY_EXISTS))
         {
-            /* If already started, find its window */
-            HWND hWindow = FindWindowW(szWindowClass, NULL);
-
-            /* Activate window */
-            ShowWindow(hWindow, SW_SHOWNORMAL);
-            SetForegroundWindow(hWindow);
+           /* If already started, find its window */
+            HWND hWindow;
+            for (int wait = 2500, inter = 250; wait > 0; wait -= inter)
+                if ((hWindow = FindWindowW(szWindowClass, NULL)) != 0)
+                    break;
+                else
+                    Sleep(inter);
+
+            /* Activate the window in the other instance */
+            SwitchToThisWindow(hWindow, TRUE);
             if (bAppwizMode)
                 PostMessage(hWindow, WM_COMMAND, ID_ACTIVATE_APPWIZ, 0);
```

Comment:
Are these correct if `hWindow == NULL`?

Reply:
Correct is a strong word, they will fail do nothing. The code was already like this but I suppose it could be better.